src/Fcf/Data/List.hs

```diff
@@ -205,3 +208,78 @@ type instance Eval (Unzip as) = Eval (Foldr Cons2 '( '[], '[]) (Eval as))
 data Cons2 :: (a, b) -> ([a], [b]) -> Exp ([a], [b])
 type instance Eval (Cons2 '(a, b) '(as, bs)) = '(a ': as, b ': bs)
 
+
+-- Helper for the Take. This corresponds to the unsafeTake in the data list lib.
+data UtakeStop :: Nat -> [a] -> Exp [a]
+type instance Eval (UtakeStop 0 _)         = '[]
+type instance Eval (UtakeStop 1 (a ': as)) = '[a]
+
+-- Helper for the Take. This corresponds to the unsafeTake in the data list lib.
+data Utake :: Nat -> [a] -> Exp [a]
+-- As the following refuses to compile, we use UtakeStop-helper.
+-- type instance Eval (Utake 0 (a ': as)) = '[]
+-- type instance Eval (Utake 1 (a ': as)) = '[a]
+-- type instance Eval (Utake m (a ': as)) = Eval ('(:) a <$> (Utake (m TL.- 1) as))
+type instance Eval (Utake n (a ': as)) = Eval
+  (If (Eval (n < 2))
+    (UtakeStop n (a ': as))
+    ('(:) a <$> (Utake (n TL.- 1) as))
+  )
+
+-- | Type-level list take.
+--
+-- === Example
+--
+-- @
+-- :kind! Eval (Take 2 '[1,2,3,4,5])
+-- @
+data Take :: Nat -> [a] -> Exp [a]
+type instance Eval (Take n as) = Eval
+  (If (Eval (n > 0))
```

Comment:
This `If` is redundant, as well as in `UtakeStop`, because the same logic can already be found in `Utake`. Given the closed type family I mentioned (and no other auxiliary definition), the `Eval` instance for `Take` should be a one-liner.

Reply:
Same goes with this drop version, there is also the "report version". I'll change this to follow that one.

Reply:
changed now